backend/packages/Upgrade/src/api/services/MoocletExperimentService.ts

```diff
@@ -231,24 +264,549 @@ export class MoocletExperimentService extends ExperimentService {
+          message: '[Mooclet Edit] Ineligible changes detected for an active Mooclet experiment.',
+          changes: allowedInactiveStateChanges,
+        };
+        logger.error({ error: JSON.stringify(error) });
+        throw new Error(JSON.stringify(error));
+      }
+
+      logger.debug({
+        message: '[Mooclet Edit] Experiment changes affect Mooclet resources, begin sync.',
+        changes: allowedInactiveStateChanges,
+      });
+
+      const currentExperiment = await this.experimentRepository.findOne({
+        where: { id: incomingExperiment.id },
+        relations: ['conditions'],
+      });
+
+      // if the changes are allowed, go ahead and update the upgrade experiment first so we can abort early if there are any issues
+      const updatedExperiment = await super.update(incomingExperiment, currentUser, logger, manager);
+
+      // ---------- update Mooclet Resources ------------------------------------------
+      // create a reference of successful changes to rollback in case of error
+      // after each successful update to mooclet resource, update this ref with the resource that was changed
+      rollbackRef.currentExperiment = currentExperiment;
+      rollbackRef.currentMoocletExperimentRef = currentMoocletExperimentRef;
+
+      // go ahead and PUT policy parameters whenever editing in enrolling or inactive, not worth detecting changes
+      if (
+        incomingExperiment.state === EXPERIMENT_STATE.INACTIVE ||
+        incomingExperiment.state === EXPERIMENT_STATE.ENROLLING
+      ) {
+        logger.debug({
+          message: '[Mooclet Edit] Upserting policy parameters due to experiment design change',
+          incomingPolicyParameters: incomingExperiment.moocletPolicyParameters,
         });
-        const map = await manager.save(MoocletVersionConditionMap, versionConditionMapEntity);
+        const policyParametersResponse = await this.handleUpdatePolicyParameters(
+          incomingExperiment.moocletPolicyParameters,
+          currentMoocletExperimentRef,
+          logger
+        );
+
+        updatedExperiment.moocletPolicyParameters = policyParametersResponse.parameters;
+        rollbackRef.revertPolicyParameters = currentPolicyParametersResponse.parameters;
+      }
+
+      // handle eligible inactive state changes one by one, else return the updated experiment early
+      // after each successful update, update the rollbackRef with the resource that was changed
+
+      if (!allowedInactiveStateChanges) {
         logger.info({
-          message: 'MoocletVersionConditionMap created successfully:',
-          versionConditionMap: JSON.stringify(map),
+          message: '[Mooclet Edit] No Mooclet-effecting experiment design changes detected.',
         });
+        return updatedExperiment;
       }
+
+      const { newOutcomeVariableName, addedConditions, removedConditions, modifiedConditions } =
+        allowedInactiveStateChanges;
+
+      if (newOutcomeVariableName) {
```

Comment:
This is happening serially. We can make them parallel. (Not that important if it affects code readability)